integrations/micrometer/src/main/java/io/helidon/integrations/micrometer/MeterRegistryProducer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.integrations.micrometer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import io.helidon.common.LazyValue;
+import io.helidon.config.Config;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+@ApplicationScoped
+final class MeterRegistryProducer {
+
+    static final String CONFIG_KEY = "micrometer";
+
+    /*
+     * Also maintains a lazy refc to the single {@code MicrometerSupport} instance.
+     */
+    private static final LazyValue<MicrometerSupport> MICROMETER_SUPPORT_FACTORY =
+            LazyValue.create(MeterRegistryProducer::createMicrometerSupport);
+
+    private MeterRegistryProducer() {
```

Comment:
I think this class is [unproxyable](https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html#unproxyable) (and you have it in a scope where it is technically required to be proxyable)

Reply:
I made the class non-`final` and made its no-args constructor `public`. I think those were the only "unproxiabilities" on it.